### ISSUE TITLE
Add alias for IPv4 rfc1918 private addresses to default OPNsense installation

### ISSUE THREAD

**Issue by nzkiwi68** [opnsense/core#4949]
**Important notices**

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues and I am convinced that mine is new.

**Is your feature request related to a problem? Please describe.**

No.

**Describe the solution you like**

Please add an alias to the default installation of OPNsense that covers private IPv4 address space.

**New Alias**
Name: rfc1918
Type: Networks
Content: 192.168.0.0/16 10.0.0.0/8 172.16.0.0/12 100.64.0.0/10 127.0.0.0/8
Description: IPv4 Private Addresses

![rfc1918](https://user-images.githubusercontent.com/67399588/116154634-085ba700-a73d-11eb-867a-3c8ea5d216b0.PNG)

**Additional context**

This is an alias that needs to be created on many installations and used with outbound firewall rules to ensure that the rule only permits traffic to the internet and not other internal protected subnets or VPN networks.

**Comment by fichtner** [comment]
Hi,

Sounds easy enough for a pull request to the default config.xml file. I will leave this open for a contributor.

Cheers,
Franco

**Comment by IsaacFL** [comment]
Doesn't the bogons alias already provide this?

**Comment by nzkiwi68** [comment]
Inbound yes.

But!!!!

**Consider this situation;**
LAN: 192.168.0.1 /24
DMZ: 192.168.88.1 /24
WAN: 202.202.202.202

Now, you want to allow a DMZ computer to access the internet for HTTPS traffic out.
If you write a firewall rule that says;
**allow src="DMZ subnet"  to dst="any" for dst port="HTTPS"**

You have actually allowed the DMZ host to access any LAN computer on HTTPS too.
A more correct firewall rule is;
**allow src="DMZ subnet"  to !dst="rfc1918" for dst port="HTTPS"**

I frequently see many firewall installations with dst=any rules that have inadvertently allowed traffic to flow into LANs, VPN tunnels, DMZ etc by mistake.

**Comment by nzkiwi68** [comment]
Bogons doesn't really work, I just had a look and it's not rfc1918
![bogons](https://user-images.githubusercontent.com/67399588/116639109-d6a44380-a9bb-11eb-9e28-09bbd2415277.PNG)

**Comment by mimugmail** [comment]
But no matter if the alias is already there or you have to add one, the Firewall rules with inverted dst wont be added by not advanced admin

**Comment by IsaacFL** [comment]
> 
> 
> Inbound yes.
> 
> But!!!!
> 
> **Consider this situation;**
> LAN: 192.168.0.1 /24
> DMZ: 192.168.88.1 /24
> WAN: 202.202.202.202
> 
> Now, you want to allow a DMZ computer to access the internet for HTTPS traffic out.
> If you write a firewall rule that says;
> **allow src="DMZ subnet" to dst="any" for dst port="HTTPS"**
> 
> You have actually allowed the DMZ host to access any LAN computer on HTTPS too.
> A more correct firewall rule is;
> **allow src="DMZ subnet" to !dst="rfc1918" for dst port="HTTPS"**
> 
> I frequently see many firewall installations with dst=any rules that have inadvertently allowed traffic to flow into LANs, VPN tunnels, DMZ etc by mistake.

What I do, is before allow to src "DMZ net"  to destination "any" would be a block rule with destination "bogons". 

I do this for all subnets, as it is never proper to allow bogons to be routed out.

**Comment by nzkiwi68** [comment]
Sure, but, bogons is not functionally equivalent to rfc1918 so your comments on what you do are great, but, not on topic.

**Comment by nzkiwi68** [comment]
> 
> 
> But no matter if the alias is already there or you have to add one, the Firewall rules with inverted dst wont be added by not advanced admin

I don't disagree with you, however, any decent admin working for business and not just simple home firewalls should always be writing proper firewall rules that never have a destination of _any_.

Perhaps OPNsense documentation could explain that with a good picture and example how that _any_ as destination is almost always not a good idea.

**Comment by OleSta** [comment]
RFC1819 is only:
10.0.0.0        -   10.255.255.255  (10/8 prefix)
172.16.0.0      -   172.31.255.255  (172.16/12 prefix)
192.168.0.0     -   192.168.255.255 (192.168/16 prefix)

not also 
100.64.0.0/10
127.0.0.0/8

**Comment by OleSta** [comment]
Is it just like this that need to be added to default config.xml?

<OPNsense>
  <Firewall>
    <Alias version="1.0.0">
      <aliases>
          <alias uuid="bd6a7585-2c83-4204-8ef8-a68680c52e33">
          <enabled>1</enabled>
          <name>RFC1918</name>
          <type>network</type>
          <proto/>
          <counters>0</counters>
          <updatefreq/>
          <content>192.168.0.0/16 172.16.0.0/12 10.0.0.0/8</content>
          <description>RFC local networks</description>
          </alias>
      </aliases>
    </Alias>
  </Firewall>
</opnsense>

**Comment by fichtner** [comment]
If you create the alias you can see the resulting config.xml entry and that is just what needs to be added to the default config.

**Comment by OleSta** [comment]
![image](https://user-images.githubusercontent.com/4288132/116994136-a1a83180-acd8-11eb-9438-8396ccdd3e8c.png)

**Comment by OleSta** [comment]
what about the uuid?
Is core/src/etc/config.xml.sample the default file?

**Comment by fichtner** [comment]
Not sure about uuid and version, @AdSchellevis knows best

Yes, that's the correct file.

**Comment by nzkiwi68** [comment]
> 
> 
> RFC1819 is only:
> 10.0.0.0 - 10.255.255.255 (10/8 prefix)
> 172.16.0.0 - 172.31.255.255 (172.16/12 prefix)
> 192.168.0.0 - 192.168.255.255 (192.168/16 prefix)
> 
> not also
> 100.64.0.0/10
> 127.0.0.0/8

True, but there is still a number of ranges of IPv4 that should never leave your network. Perhaps for more a correct name we call this alias;
**PrivateIPv4**

And include;
10.0.0.0/8           - rfc1918
172.16.0.0/12     - rfc1918
192.168.0.0/16   - rfc1918
192.0.2.0/24       - TEST-NET-1 rfc5737
198.51.100.0/24  -TEST-NET-1 rfc5737
203.0.113.0/24    -TEST-NET-1 rfc5737
198.18.0.0/15      - Benchmark testing rfc2544
100.64.0.0/10     - Shared Address Space rfc6598

**References:**
[https://tools.ietf.org/html/rfc5737](url)
[https://tools.ietf.org/html/rfc2544](url)
[https://tools.ietf.org/html/rfc6598](url)

**Comment by AdSchellevis** [comment]
@fichtner @OleSta  when adding an entry to the default config, one could generate a uuid and just slip it in (which would make it equal on all setups). Another option might be to add an entry to the reserved items, if I'm not mistaken these will just be flushed when applied. (but will likely overwrite existing entries with the same name)

https://github.com/opnsense/core/blob/af587e812a7b8d96cf3fd84869dc7e00bbfe09b3/src/opnsense/mvc/app/models/OPNsense/Firewall/FieldTypes/AliasField.php#L37-L66

Adding a container concept like "PrivateIPv4" has the risk of being somewhat diffuse again (different purpose), I think we should be careful adding standard networks in an aggregated form again.

**Comment by OleSta** [comment]
PrivateIPv4 container should not be added.
WAN interface setup should take care of not routning the reserved networks:
![image](https://user-images.githubusercontent.com/4288132/117106804-cbb52e80-ad80-11eb-9763-73f23cbdea06.png)

Adding only rfc1918 networks should be fine for the DMZ example in top.
(With only one other internal network you can just make the rule: !dst="LAN network")
All other networks should not be in use internal.

**Comment by OleSta** [comment]
Bogons lists and blocks the rest of the list and prevent them from beeing routed:
![image](https://user-images.githubusercontent.com/4288132/117107450-e50aaa80-ad81-11eb-84b6-716ad05ffde3.png)

**Comment by fichtner** [comment]
My preference would be to avoid UUID for default entries and render one when first read/migrated (so no version number as well). Contents should be editable, but only carry actual RFC private content indeed for defaults. We came along way without them so no need to overcomplicate this.

With new alias types added in 21.7 it may be reasonable to discuss other automated ones, e.g. generating aliases from interface contents (multi-select) which is likely better suited for the actual requirement laid out here.

In any case, manual effort will always be involved. This should be factored into implementations and documentation to avoid false hopes.

**Comment by nzkiwi68** [comment]
> 
> 
> Bogons lists and blocks the rest of the list and prevent them from beeing routed:
> ![image](https://user-images.githubusercontent.com/4288132/117107450-e50aaa80-ad81-11eb-84b6-716ad05ffde3.png)

I don't think your statement is correct. the bogon list blocks inbound packets via an auto generated rule on the WAN interface, true, but, since OPNsense is a stateful firewall, an outbound packet will be permitted to leave the firewall destined to a bogon network and as such will have a valid state that allows an answer from the WAN despite the bogon block rule.
